termux-shared/src/main/java/com/termux/shared/shell/ShellUtils.java

```diff
@@ -0,0 +1,188 @@
+
+        List<String> environment = new ArrayList<>();
+
+        // This function may be called by a different package like a plugin, so we get version for Termux package via its context
+        Context termuxPackageContext = TermuxUtils.getTermuxPackageContext(currentPackageContext);
+        if (termuxPackageContext != null) {
+            String termuxVersionName = PackageUtils.getVersionNameForPackage(termuxPackageContext);
+            if (termuxVersionName != null)
+                environment.add("TERMUX_VERSION=" + termuxVersionName);
+        }
+
+        environment.add("TERM=xterm-256color");
+        environment.add("COLORTERM=truecolor");
+        environment.add("HOME=" + TermuxConstants.TERMUX_HOME_DIR_PATH);
+        environment.add("PREFIX=" + TermuxConstants.TERMUX_PREFIX_DIR_PATH);
+        environment.add("BOOTCLASSPATH=" + System.getenv("BOOTCLASSPATH"));
+        environment.add("ANDROID_ROOT=" + System.getenv("ANDROID_ROOT"));
+        environment.add("ANDROID_DATA=" + System.getenv("ANDROID_DATA"));
+        // EXTERNAL_STORAGE is needed for /system/bin/am to work on at least
+        // Samsung S7 - see https://plus.google.com/110070148244138185604/posts/gp8Lk3aCGp3.
+        environment.add("EXTERNAL_STORAGE=" + System.getenv("EXTERNAL_STORAGE"));
+
+        // These variables are needed if running on Android 10 and higher.
+        addToEnvIfPresent(environment, "ANDROID_ART_ROOT");
+        addToEnvIfPresent(environment, "DEX2OATBOOTCLASSPATH");
+        addToEnvIfPresent(environment, "ANDROID_I18N_ROOT");
+        addToEnvIfPresent(environment, "ANDROID_RUNTIME_ROOT");
+        addToEnvIfPresent(environment, "ANDROID_TZDATA_ROOT");
+
+        if (isFailSafe) {
+            // Keep the default path so that system binaries can be used in the failsafe session.
+            environment.add("PATH= " + System.getenv("PATH"));
+        } else {
+            environment.add("LANG=en_US.UTF-8");
+            environment.add("PATH=" + TermuxConstants.TERMUX_BIN_PREFIX_DIR_PATH);
+            environment.add("PWD=" + workingDirectory);
+            environment.add("TMPDIR=" + TermuxConstants.TERMUX_TMP_PREFIX_DIR_PATH);
+        }
+
+        return environment.toArray(new String[0]);
+    }
+
+    public static void addToEnvIfPresent(List<String> environment, String name) {
+        String value = System.getenv(name);
+        if (value != null) {
+            environment.add(name + "=" + value);
+        }
+    }
+
+    public static int getPid(Process p) {
+        try {
+            Field f = p.getClass().getDeclaredField("pid");
+            f.setAccessible(true);
+            try {
+                return f.getInt(p);
+            } finally {
+                f.setAccessible(false);
+            }
+        } catch (Throwable e) {
+            return -1;
+        }
+    }
+
+    public static String[] setupProcessArgs(@NonNull String fileToExecute, String[] arguments) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.